### ISSUE TITLE
variables referenced without `var.` fail silently

### ISSUE THREAD

**Issue by MarkHerhold** [hashicorp/terraform#8010]
As a new user, I didn't realize I had to use `var.` to reference a variable as in `"${var.my-service-name}"`. This caused a lot of issues for me such as security group rules not being created, tags not being created, etc.
### Terraform Version

Terraform v0.7.0
(I also had this issue in 0.6.14)
### Affected Resource(s)

Potentially everything that you can reference a variable in.
Specifically, I experienced issues using improperly referenced variables in aws_subnet, aws_security_group.
### Expected Behavior

I would expect `terraform plan` to throw an error and `terraform apply` to completely fail during execution.

There should be an error stating "Cannot reference `my-sg-name` in main.tf at 55:10"
### Actual Behavior

Everything shows up as computed and tags and other fields simply are not created.

Part of `mail.tf`:

```
resource "aws_security_group" "servers" {
  name        = "servers_sg"
  vpc_id      = "${aws_vpc.vpc.id}"
...
  tags {
    "Name" = "${my-sg-name}"
  }
```

Output:

```
~ aws_subnet.servers
    tags.%: "" => "<computed>"
```
### References

See @Zapadlo 's comment in GH-6265 about variables failing silently

**Comment by mitchellh** [comment]
Talked this over with @phinze and I'm going to tag this with "0.8":

We'd like to break some backwards compatibility to fix this the right way. The "right way" we believe is to make it so that `template_file` `source` parameter must be escaped if its a raw literal. The core ignores unprefixed variables (known internally as "simple" variables) _only_ for this case and we think that wasn't the right decision.

However, this is a BC-incompat change so we will hold it until 0.8.

**Comment by mitchellh** [comment]
The fix for this is in #9698 and HCL. This will be part of 0.8. Closing this as a dup though of #5338 realized it was there as well.
